.travis.yml

```diff
@@ -1,17 +1,68 @@
+sudo: false
 language: cpp
 
 compiler:
   - gcc
-  - clang
 
 env:
+  matrix:
+    - FUZZ=1
+    - NO_INLINE=1
+    - NO_BIT=1
+    - STATIC=1 NO_DEBUG=1
+    - SAFETY_PROFILE=1
+    - MOD_DEFAULT_FEATURES=1 USES_WCHAR=1
   global:
-    - ACE_ROOT=$TRAVIS_BUILD_DIR/ext/ACE_TAO/ACE
-    - TAO_ROOT=$TRAVIS_BUILD_DIR/ext/ACE_TAO/TAO
+    - ACE_ROOT=$TRAVIS_BUILD_DIR/ACE_TAO/ACE
+    - TAO_ROOT=$TRAVIS_BUILD_DIR/ACE_TAO/TAO
     - DDS_ROOT=$TRAVIS_BUILD_DIR
     - MPC_ROOT=$TRAVIS_BUILD_DIR/ext/MPC
     - LD_LIBRARY_PATH=$ACE_ROOT/lib:$DDS_ROOT/lib:$LD_LIBRARY_PATH
 
+matrix:
+  include:
+    - compiler: gcc
+      addons:
+        apt:
+          packages:
+            - openjdk-7-jdk
+      env: FUZZ=1 USE_JAVA=1
+    - compiler: clang
+    - compiler: gcc
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+          packages:
+            - g++-4.9
+      env: USE_ALT_COMPILER=1 COMPILER=g++-4.9
+    - compiler: gcc
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+          packages:
+            - g++-5
+      env: USE_ALT_COMPILER=1 COMPILER=g++-5
+    - compiler: clang
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+            - llvm-toolchain-precise-3.6
+          packages:
+            - clang++-3.6
+      env: USE_ALT_COMPILER=1 COMPILER=clang++-3.6
```

Comment:
I don't think we need to test with Clang 3.6 if we also have a newer Clang working.